### PR TITLE
Adds grille functionality to wall frames

### PR DESCRIPTION
**Pull request by CapybaraExtravagante**
Adds the shocking properties of grilles over to window frames.

Some gameplay differences:

- Bumping does not shock
- Punching does shock, I'm not sure if this is something we wants, it depends on if we want you to be able to punch out grilles or not.
- Climbing onto wallframes will shock you 100% of the time.
- Getting onto wallframes will shock you, except when flying.

I also moved some tool stuff to the tool procs.

### CODE REVIEW

File: code/game/turfs/closed/wall/window_frames.dm [hunk 1]
```diff
@@ -39,11 +39,16 @@
...
-	AddElement(/datum/element/climbable)
+	AddElement(/datum/element/climbable, on_try_climb_procpath = TYPE_PROC_REF(/obj/structure/window_frame/, on_try_climb))
 
 	if(mapload && start_with_window)
 		create_structure_window(window_type, TRUE)
 
+	var/static/list/loc_connections = list(
+		COMSIG_ATOM_ENTERED = PROC_REF(on_entered),
+	)
+	AddElement(/datum/element/connect_loc, loc_connections)
```
Comment (ArcaneMusic) [inline code/game/turfs/closed/wall/window_frames.dm:50]: Could you test if this means throwing a player over a low wall would shock them as they pass over? Ideally if they're being thrown over and flying like that they wouldn't be shocked

File: code/game/turfs/closed/wall/window_frames.dm [hunk 2]
```diff
@@ -54,6 +59,103 @@
...
+/obj/structure/window_frame/proc/shock(mob/user, shock_chance)
+	if(!has_grille) // no grille? dont shock.
+		return FALSE
+	if(!prob(shock_chance))
+		return FALSE
+	if(!in_range(src, user))//To prevent TK and mech users from getting shocked
+		return FALSE
+	var/turf/my_turf = get_turf(src)
+	var/obj/structure/cable/underlaying_cable = my_turf.get_cable_node()
+	if(underlaying_cable)
```
Comment (ArcaneMusic) [inline code/game/turfs/closed/wall/window_frames.dm]: Check `!underlaying_cable` at the top of the proc for an early return, saves some time

**Review by ArcaneMusic — COMMENTED** [review]
All in all looks good, lets just look over these two things and we're probably good

File: code/game/turfs/closed/wall/window_frames.dm (hunk 1 above)
Comment (CapybaraExtravagante) [reply code/game/turfs/closed/wall/window_frames.dm:50]: it  probably will! good point :) I'll check for the flags

Comment (CapybaraExtravagante) [reply code/game/turfs/closed/wall/window_frames.dm:50]: Apparently I already do this :P 

```/obj/structure/window_frame/proc/on_entered(datum/source, AM as mob|obj)
	SIGNAL_HANDLER
	if(!isliving(AM))
		return
	var/mob/living/potential_victim = AM
	if(potential_victim.movement_type & (FLOATING|FLYING))
		return
	shock(potential_victim, 100)```

File: code/game/turfs/closed/wall/window_frames.dm [hunk 3]
```diff
@@ -54,6 +59,102 @@
 
 	return FALSE
 
+///Called by the climbable element if you try climb up. Better hope you're well protected against shocks! XD
+/obj/structure/window_frame/proc/on_try_climb(mob/climber)
+	shock(climber, 100)
+
+///Gives the user a shock if they get unlucky (Based on shock chance)
+/obj/structure/window_frame/proc/shock(mob/user, shock_chance)
```
Comment (LemonInTheDark) [inline code/game/turfs/closed/wall/window_frames.dm]: call this try_shock, s more clear

File: code/game/turfs/closed/wall/window_frames.dm [hunk 4]
```diff
@@ -54,6 +59,102 @@
...
+	var/turf/my_turf = get_turf(src)
+	var/obj/structure/cable/underlaying_cable = my_turf.get_cable_node()
+	if(electrocute_mob(user, underlaying_cable, src, 1, TRUE))
+		var/datum/effect_system/spark_spread/spark_effect = new /datum/effect_system/spark_spread
+		spark_effect.set_up(3, 1, src)
+		spark_effect.start()
+		return TRUE
+	return FALSE
+
+/obj/structure/window_frame/proc/on_entered(datum/source, AM as mob|obj)
```
Comment (LemonInTheDark) [inline code/game/turfs/closed/wall/window_frames.dm]: don't use as, it's a legacy system used for verbs. just type it to atom movable

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]

File: code/game/turfs/closed/wall/window_frames.dm (hunk 4 above)
Comment (CapybaraExtravagante) [reply code/game/turfs/closed/wall/window_frames.dm]: ah, i copy-pasted this proc from somewhere else, I'll change it.

**Pull request merged**